src/pip/_internal/network/lazy_wheel.py

```diff
@@ -204,7 +265,57 @@ def _download(self, start: int, end: int) -> None:
             right = bisect_right(self._left, end)
             for start, end in self._merge(start, end, left, right):
                 response = self._stream_response(start, end)
-                response.raise_for_status()
                 self.seek(start)
-                for chunk in response_chunks(response, self._chunk_size):
+                for chunk in response.iter_content(self._chunk_size):
                     self._file.write(chunk)
+
+    def prefetch(self, target_file: str) -> None:
+        """
+        Prefetch a specific file from the remote ZIP in one request.
+        """
+        with self._stay():  # not strictly necessary
+            # try to read entire conda info in one request
+            zf = ZipFile(self)  # type: ignore
+            infolist = zf.infolist()
+            for i, info in enumerate(infolist):
+                if info.filename == target_file:
+                    # could be incorrect if zipfile was concatenated to another
+                    # file (not likely for .conda)
+                    start = info.header_offset
+                    try:
+                        end = infolist[i + 1].header_offset
+                        # or info.header_offset
+                        # + len(info.filename)
+                        # + len(info.extra)
+                        # + info.compress_size
+                        # (unless Zip64)
+                    except IndexError:
+                        end = zf.start_dir
+                    self.seek(start)
+                    self.read(end - start)
+                    log.debug(
+                        "prefetch %s-%s",
+                        info.header_offset,
+                        end,
+                    )
+                    break
+            else:
+                log.debug("no zip prefetch")
+
+    def prefetch_dist_info(self) -> None:
+        """
+        Read contents of entire dist-info section of wheel.
+
+        pip wants to read WHEEL and METADATA.
+        """
+        with self._stay():
+            zf = ZipFile(self)  # type: ignore
+            infolist = zf.infolist()
+            for info in infolist:
+                # should be (wheel filename without extension etc) + (.dist-info/)
+                if ".dist-info/" in info.filename:
```

Comment:
I think there’s a better detection logic somewhere (probably in wheel installation code) for this.

Reply:
In pypa/wheel wheelfile.py it derives the .dist-info dirname from the parsed wheel filename. https://github.com/pypa/wheel/blob/main/src/wheel/wheelfile.py#L49, there probably is similar code in pip.

Reply:
If there is similar code existing in the code base, we should extract the common logic.